src/main/java/com/monumental/services/MonumentService.java

```diff
@@ -100,48 +129,33 @@ private void buildDWithinQuery(CriteriaBuilder builder, CriteriaQuery query, Roo
     @SuppressWarnings("unchecked")
     public List<Monument> search(String searchQuery, String page, String limit, String latitude, String longitude,
                                  String distance) {
-
         CriteriaBuilder builder = this.getCriteriaBuilder();
         CriteriaQuery<Monument> query = this.createCriteriaQuery(builder, false);
         Root<Monument> root = this.createRoot(query);
         root.fetch("tags", JoinType.LEFT);
         query.select(root);
 
-        // TODO: Query for description
-        // TODO: Query for tags
-        // TODO: Filters
-
-        if (searchQuery != null) {
-            this.buildSimilarityQuery(builder, query, root, searchQuery, 0.1, true);
-        }
-
-        if (latitude != null && longitude != null && distance != null) {
-            this.buildDWithinQuery(builder, query, root, latitude, longitude, Integer.parseInt(distance));
-        }
+        this.buildSearchQuery(builder, query, root, searchQuery, latitude, longitude, distance);
 
-        List<Monument> monuments = this.getWithCriteriaQuery(query, Integer.parseInt(limit), (Integer.parseInt(page)) - 1);
+        List<Monument> monuments = limit != null
```

Comment:
Not sure I love the nested ternaries lol

Reply:
😍